### PR TITLE
ref: Kill the last remains of the legacy message use internally

### PR DESCRIPTION
**Pull request by mitsuhiko**
This also changes how `Event.message` is generated. It now does it for all event types because we now only look at logentry message anyways.

#sync-getsentry

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -78,6 +78,20 @@
 )
 
 
+def get_event_metadata_compat(data, fallback_message):
+    """This is a fallback path to getting the event metadata.  This is used
+    by some code paths that could potentially deal with old sentry events that
+    do not have metadata yet.  This does not happen in practice any more but
+    the testsuite was never adapted so the tests hit this code path constantly.
+    """
```
Comment (markstory) [inline src/sentry/event_manager.py]: Should we update the test suite?

Comment (mitsuhiko) [reply src/sentry/event_manager.py]: I will do this in a followup PR.

**Comment by mitsuhiko** [comment]
We can either merge this as such or I do part two which renames `Event.message` to `Event.search_message` and cleans up some other stuff in the process. I'm in favor of the latter and get this merged on monday.

File: src/sentry/templates/sentry/emails/activity/generic.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 ## Issue Details
 
-{{ group.message_short }}
+{{ group.title }}
```
Comment (jan-auer) [inline src/sentry/templates/sentry/emails/activity/generic.txt]: Shouldn't this become `group.message` now?

Comment (mitsuhiko) [reply src/sentry/templates/sentry/emails/activity/generic.txt]: to quote the existing code:

```
    @property
    def message_short(self):
        warnings.warn('Group.message_short is deprecated, use Group.title', DeprecationWarning)
        return self.title
```

**Comment by mitsuhiko** [comment]
The test failure is just codedov being nasty. I think this should merge after review.

**Review by jan-auer — APPROVED** [review]
🤞

**Pull request merged**